src/app/components/toggle/toggle.component.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  AfterViewInit,
+  Component,
+  CUSTOM_ELEMENTS_SCHEMA,
+  ElementRef,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+
+@Component({
+  selector: 'app-toggle',
+  standalone: true,
+  template: `
+    <h2 class="tds-headline-02 tds-u-pb1">Toggle</h2>
+    <tds-toggle
+      toggle-id="example-toggle"
+      headline="Test this toggle, observe Console for results"
+      size="lg"
+      #toggleComponent
+    >
+      <div slot="label">Click on me!</div>
+    </tds-toggle>
+  `,
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
+})
+export class ToggleComponent implements AfterViewInit {
+  @ViewChild('toggleComponent', { static: true }) toggleComponent: ElementRef;
+
+  constructor(private renderer: Renderer2) {}
+
+  ngAfterViewInit() {
+    const webComponent = this.toggleComponent.nativeElement;
```

Comment:
are there any reason why we select the element attach event listener here? 

for example if we remove:
 -#toggleComponent selector  
   - @ViewChild('toggleComponent', { static: true }) toggleComponent: ElementRef;
   - renderer2 injection
   
   
  and just use (tdsToggle)="listenme()" In the template. this should be enough?

Reply:
I agree, normally you do event binding directly in the template and you bind it to a function in the component.ts file.

Reply:
Yep, you are right. I was looking into different options, but binding it directly in the template seems the best and most simple. I think our issue is to properly pass event types in Angular, one way is to list them down in types.ds

Reply:
yes, you can check it out here: https://angular.io/guide/event-binding